Fixes #8934: Allow plugin menu items to disable turbolinks for their lin...

ehelms opened the pull request:
...ks.

ehelms committed:
Fixes #8934: Allow plugin menu items to disable turbolinks for their links.

ehelms commented:
Please see https://github.com/theforeman/foreman/pull/2081 to test this.
Pre-patch
1. Setup Foreman + Katello
2. Start server.
3. Go to 'Products' page.
4. Note blank page.
5. Stop server.
6. Remove tmp/

Post-patch
1. Apply both of these patches.
2. Start server.
3. Go to 'Products' page.
4. See products page.

dLobatog commented:
@ehelms Looking good, I'll test it tomorrow morning my time but can you add a integration test for Katello that checks these kind of interactions work (said test should fail until this is merged)?

ohadlevy commented on app/services/menu/item.rb:
```diff
@@ -1,6 +1,6 @@
 module Menu
   class Item < Node
-    attr_reader :name, :condition, :parent, :child_menus, :last, :html_options
+    attr_reader :name, :condition, :parent, :child_menus, :last, :html_options, :no_turbolink
```
can turbolinks be true false, double negation is confusing ?

ehelms commented:
@eLobato Updated. Added a few Foreman tests and added a test on the Katello side that checks that the menu items created have turbolinks set to false on them.

ohadlevy commented:
merged as 1a36b54 thanks @ehelms
